warp10/src/main/java/io/warp10/script/ext/shm/MUTEX.java

```diff
@@ -54,6 +52,10 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
 
     String mutex = String.valueOf(top);
 
+    if(null != stack.getAttribute(MUTEX_ATTRIBUTE + stack.getUUID()) && mutex != stack.getAttribute(MUTEX_ATTRIBUTE + stack.getUUID())) {
```

Comment:
Should be `==`

Reply:
The condition has been changed to use equals() but it is correct in the sense that we want to detect nested calls to MUTEX with different mutex names.